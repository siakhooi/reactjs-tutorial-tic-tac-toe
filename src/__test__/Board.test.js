import React from 'react'
import renderer from 'react-test-renderer'
import Board from '../Board'

it('Board', () => {
  const dummyFunction = jest.fn()
  const squares = Array(9).fill(null)
  const component =
    renderer.create(
      <Board
        squares={squares}
        onClick={dummyFunction}
      />
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.children.forEach((x) => {
    x.children.forEach((y) => {
      y.props.onClick()
    })
  })
  expect(dummyFunction).toBeCalledTimes(9)
})
