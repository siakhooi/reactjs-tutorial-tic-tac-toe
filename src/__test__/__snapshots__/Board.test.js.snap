// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board 1`] = `
<div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
</div>
`;
